charts/microgateway/README.md.gotmpl

```diff
@@ -1,153 +1,365 @@
+
+This means that:
+* Only one Virtual Host is configured.
+* The Mapping configuration is applied to all requests sent to the backend service.
+* There is only one backend service.
+
+
 
-### Default DSL
 The Helm chart provides a simple configuration which can be configured with `config.default.*` parameters.
 All settings have already configured a default value. So only the values which differ from the default value have to be configured. 
 
-Example code fragment:
-Customization of the backend hostname.
+
+**Example:**
+
 custom-values.yaml
 ```
 config:
+  IPHeader:
+    trustedProxies:
+      - 10.0.0.0/28
   default:
+    mapping:
+      entryPath: /
     backend:
       hostname: custom-backend
+
+redis:
+  enabled: true
 ```
 
-### Custom DSL App
-If the default app settings are not sufficient, you can define a custom app as YAML with the `config.apps` parameter. 
-This setting overwrites the default app (mapping & backend), but the remaining settings of the DSL can still be configured with the default DSL method. 
-Example code fragment:
-Customization of the VirtualHost with Apache Expert Settings.
+**Parameters which can be used**:
+* All `config.*` parameters can be used.
+  * When using the `config.redisService` parameter:
+    * The Mapping will be configured with "enforce_session", except `config.default.mapping.sessionHandling` is configured.
+  * When using the `redis.enabled` parameter:
+    * The Mapping will be configured with "enforce_session", except `config.default.mapping.sessionHandling` is configured.
+* All `config.default.*` parameters can be used.
+  
+
+**Parameters which cannot be used**:
+* The `config.apps` parameter cannot be used. This would switch to the [Advanced DSL app configuration](#advanced-dsl-app-configuration)
+* The `config.dsl` parameter cannot be used. This would switch to the [Expert DSL app configuration](#expert-dsl-app-configuration)
+
+
+### Advanced DSL app configuration
+
+In case that the [Simple DSL configuration](#simple-dsl-configuration) does not suite, the advanced configuration options might help. The following use cases might require this kind of configuration:
+
+
+Use Case 1)
+* Virtual Host 1: abc.com -> Mapping 1: / -> Backend Service 1
+* Virtual Host 2: xyz.com -> Mapping 2: / -> Backend Service 1
+
+Use Case 2)
+* Virtual Host 1: abc.com -> Mapping 1: /      -> Backend Service 1
+                          -> Mapping 2: /auth/ -> Backend Service 1
+
+Use Case 3)
+* Virtual Host 1: abc.com -> Mapping 1: / -> Backend Service 1
+* Virtual Host 2: xyz.com -> Mapping 2: / -> Backend Service 2
+
+These use cases are only examples which could also occur slightly different. But all of them have in common that they have more than one Virtual Hosts, Mappings or Backend Services. When ever this is the case, the [Advanced DSL app configuration](#advanced-dsl-app-configuration) should be preferred over the [Simple DSL configuration](#simple-dsl-configuration).
```

Comment:
When ever --> Whenever

Reply:
should be preferred over the
It is not possible to configure it with the simple setup. 
So I would use. 
must be used instead of the

Reply:
Fixed.